### PR TITLE
fix: cluster size of 0 leads to infinite loop

### PR DESCRIPTION
**Pull request by fResult**
## Summary

Fixes infinite loop bug when cluster() size is 0.
I chose error throwing over returning an input array due to `Cluster<T, Size>` type constraints.

Additional test cases were added to verify handling of various edge cases, including negative and non-integer sizes.

## Related issue, if any:

https://github.com/radashi-org/radashi/issues/392

## For any code change,

- [x] Related documentation has been updated, if needed
- [x] Related tests have been added or updated, if needed
- [x] Related benchmarks have been added or updated, if needed
- [x] Release notes in [next-minor.md](.github/next-minor.md) or [next-major.md](.github/next-major.md) have been added, if needed

## Does this PR introduce a breaking change?

No

## Bundle impact

| Status | File | Size [^1337] | Difference |
| --- | --- | --- | --- |
| M | `src/array/cluster.ts` | 116 | +7 (+6%) |

[^1337]: Function size includes the `import` dependencies of the function.

### CODE REVIEW

File: src/array/cluster.ts [hunk 1]
```diff
@@ -13,6 +15,14 @@ export function cluster<T, Size extends number = 2>(
   array: readonly T[],
   size: Size = 2 as Size,
 ): Cluster<T, Size>[] {
+  if (size <= 0 || Number.isNaN(size)) {
```
Comment (aleclarson) [inline src/array/cluster.ts:18]: The NaN check is unnecessary, since the loop will exit when `size` is `NaN`.

File: src/array/cluster.ts [hunk 2]
```diff
@@ -13,6 +15,14 @@ export function cluster<T, Size extends number = 2>(
...
+    throw new IllegalSizeError(`Size must be 1 or more, the size is ${size}`)
+  }
+
+  if (size === Number.POSITIVE_INFINITY) {
```
Comment (aleclarson) [inline src/array/cluster.ts:22]: Same here. The loop will exit just fine when `size` is infinite.

File: src/array/cluster.ts [hunk 3]
```diff
@@ -13,6 +15,14 @@ export function cluster<T, Size extends number = 2>(
   array: readonly T[],
   size: Size = 2 as Size,
 ): Cluster<T, Size>[] {
+  if (size <= 0 || Number.isNaN(size)) {
+    throw new IllegalSizeError(`Size must be 1 or more, the size is ${size}`)
```
Comment (aleclarson) [inline src/array/cluster.ts:19]: I think returning `[]` when `size <= 0` would be better. It's more concise, makes logical sense, and works fine with the `Cluster<T, Size>[]` return type.

**Review by aleclarson — CHANGES_REQUESTED** [review]
Hey @fResult, good to see more contributions from you 👍 

I'll get this merged after my review is addressed

**Comment by fResult** [comment]
Hi @aleclarson, I have already addressed all suggestions.
Thanks for your review. 🙏

**Comment by aleclarson** [comment]
Appreciate the extra tests! Thanks @fResult
